ext/prism/extension.c

```diff
@@ -599,8 +599,7 @@ parse_lex_input(pm_string_t *input, const pm_options_t *options, bool return_nod
 
     VALUE source_string = rb_str_new((const char *) pm_string_source(input), pm_string_length(input));
     VALUE offsets = rb_ary_new();
-    VALUE source_argv[] = { source_string, LONG2NUM(parser.start_line), offsets };
-    VALUE source = rb_class_new_instance(3, source_argv, rb_cPrismSource);
+    VALUE source = rb_funcall(rb_cPrismSource, rb_intern("create"), 3, source_string, LONG2NUM(parser.start_line), offsets);
```

Comment:
Since we're coming from C, I think I would prefer here that we do the ascii_only check ourselves, because it will also force a coderange check, which will be nice.

That would be something like:

```c
VALUE source = rb_class_new_instance(3, source_argv, rb_str_is_ascii_only_p(source_string) ? rb_cPrismASCIISource : rb_cPrismSource);
```